packages/core/src/routes/session/utils.ts

```diff
@@ -79,64 +79,57 @@ export const checkVerificationSessionByFlowAndMedium = (
 
 export const checkAndGetUserIdFromPayload = async (
   flow: PasscodeType,
-  medium: Medium,
+  method: Method,
   payload: Pick<VerificationStorage, 'email' | 'phone'>
-): Promise<string> => {
+) => {
   const { email, phone } = payload;
 
-  // eslint-disable-next-line @silverhand/fp/no-let
-  let user: User;
-  // eslint-disable-next-line @silverhand/fp/no-let
-  let id: string;
-
-  switch (flow + medium) {
-    case PasscodeType.SignIn + 'sms':
-      assertThat(
-        phone && (await hasUserWithPhone(phone)),
-        new RequestError({ code: 'user.phone_not_exists', status: 422 })
-      );
-
-      // eslint-disable-next-line @silverhand/fp/no-mutation
-      user = await findUserByPhone(phone);
-      // eslint-disable-next-line @silverhand/fp/no-mutation
-      id = user.id;
-
-      return id;
-    case PasscodeType.SignIn + 'email':
-      assertThat(
-        email && (await hasUserWithEmail(email)),
-        new RequestError({ code: 'user.email_not_exists', status: 422 })
-      );
-
-      // eslint-disable-next-line @silverhand/fp/no-mutation
-      user = await findUserByEmail(email);
-      // eslint-disable-next-line @silverhand/fp/no-mutation
-      id = user.id;
-
-      return id;
-    case PasscodeType.Register + 'sms':
-      assertThat(
-        phone && !(await hasUserWithPhone(phone)),
-        new RequestError({ code: 'user.phone_exists_register', status: 422 })
-      );
-
-      // eslint-disable-next-line @silverhand/fp/no-mutation
-      id = await generateUserId();
-
-      return id;
-    case PasscodeType.Register + 'email':
-      assertThat(
-        email && !(await hasUserWithEmail(email)),
-        new RequestError({ code: 'user.email_exists_register', status: 422 })
-      );
-
-      // eslint-disable-next-line @silverhand/fp/no-mutation
-      id = await generateUserId();
-
-      return id;
-    default:
-      throw new Error('Not implemented');
+  if (flow === PasscodeType.SignIn && method === 'sms') {
```

Comment:
Prefer 4 routes instead of this 4-if-statements function. That is the job of "router"?